### PR TITLE
[TTAHUB-2587] Data Request - Approved Activity Reports with CLASS Goal

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change
Added two new JDI queries:

- class-goal-use.sql
- fake-class-goal-use.sql

## How to test
run the queries using the description provided in each file for how to use it.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2587

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/queries/fake-class-goal-use.sql [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+JOIN "ActivityReportGoals" arg
+ON g.id = arg."goalId"
+JOIN "ActivityReports" ar
+ON arg."activityReportId" = ar.id
+JOIN "Grants" gr
+ON g."grantId" = gr.id
+JOIN "Recipients" r
+ON gr."recipientId" = r.id
+WHERE ar."calculatedStatus" = 'approved'
+AND g.name like '%CLASS%'
```
Comment (hardwarehuman) [inline src/queries/fake-class-goal-use.sql:39]: Do we want to do an `UPPER(g.name)` here?

File: src/queries/fake-class-goal-use.sql [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+-- Filter for createdAt dates between two values if jdi.createdbetween is defined
+(NULLIF(current_setting('jdi.createdbetween', true), '') IS NULL
+      OR g."createdAt"::date <@ (
+        SELECT CONCAT('[',MIN(value::timestamp),',',MAX(value::timestamp),')')::daterange AS my_array
+        FROM json_array_elements_text(COALESCE(NULLIF(current_setting('jdi.createdbetween', true), ''),'[]')::json) AS value
+      ))
+AND
+-- Filter for startDate dates between two values if jdi.startDate is defined
+(NULLIF(current_setting('jdi.startDate', true), '') IS NULL
+      OR ar."startDate"::date <@ (
```
Comment (hardwarehuman) [inline src/queries/fake-class-goal-use.sql:96]: I had no idea this would work.

**Review by hardwarehuman — APPROVED** [review]
Except for the possible UPPER(), I think this is good to go.

File: src/queries/fake-class-goal-use.sql (hunk 1 above)
Comment (GarrettEHill) [reply src/queries/fake-class-goal-use.sql:39]: I think I want to keep case as it is as CLASS is used as a noun using its capitalization. So class and CLASS are different things.

File: src/queries/fake-class-goal-use.sql (hunk 2 above)
Comment (GarrettEHill) [reply src/queries/fake-class-goal-use.sql:96]: https://www.postgresql.org/docs/current/rangetypes.html#RANGETYPES-IO

**Review by AdamAdHocTeam — APPROVED** [review]
I didn't get into the data too much but runs without issue.

**Pull request merged**